Type annotation for `health` doesn't support async methods

```python
from fastapi import Depends, FastAPI
from fastapi_health import health

def get_session():
    return True

def is_database_online(session: bool = Depends(get_session)):
    return session

async def async_condition() -> bool:
    return True

app = FastAPI()
app.add_api_route("/health", health([is_database_online, async_condition]))
```

The above code fails MyPy check due to the anntoation on `health` not supporting async functions.

```
main.py:16: error: List item 1 has incompatible type "Callable[[], Coroutine[Any, Any, bool]]"; expected "Callable[..., dict[str, Any] | bool]"  [list-item]
Found 1 error in 1 file (checked 1 source file)
```

As outlined in #9, the library does support using async functions, so I think the annotation can be updated to add support.

--- a/fastapi_health/route.py
+++ b/fastapi_health/route.py
@@ -6,6 +6,8 @@ from fastapi.encoders import jsonable_encoder
 from fastapi.responses import JSONResponse
 
 T = TypeVar("T")
+HealthcheckReturnT = Union[Dict[str, Any], bool]
+ConditionFuncT = Callable[..., Union[HealthcheckReturnT, Awaitable[HealthcheckReturnT]]]
 
 
 async def default_handler(**kwargs: T) -> Dict[str, T]:
@@ -24,7 +26,7 @@ async def default_handler(**kwargs: T) -> Dict[str, T]:
 
 
 def health(
-    conditions: List[Callable[..., Union[Dict[str, Any], bool]]],
+    conditions: List[ConditionFuncT],
     *,
     success_handler: Callable[..., Awaitable[dict]] = default_handler,
     failure_handler: Callable[..., Awaitable[dict]] = default_handler,

--- a/fastapi_health/route.py
+++ b/fastapi_health/route.py
@@ -6,8 +6,8 @@ from fastapi.encoders import jsonable_encoder
 from fastapi.responses import JSONResponse
 
 T = TypeVar("T")
-HealthcheckReturnT = Union[Dict[str, Any], bool]
-ConditionFuncT = Callable[..., Union[HealthcheckReturnT, Awaitable[HealthcheckReturnT]]]
+HealthcheckReturn = Union[Dict[str, Any], bool]
+ConditionFunc = Callable[..., Union[HealthcheckReturn, Awaitable[HealthcheckReturn]]]
 
 
 async def default_handler(**kwargs: T) -> Dict[str, T]:
@@ -26,7 +26,7 @@ async def default_handler(**kwargs: T) -> Dict[str, T]:
 
 
 def health(
-    conditions: List[ConditionFuncT],
+    conditions: List[ConditionFunc],
     *,
     success_handler: Callable[..., Awaitable[dict]] = default_handler,
     failure_handler: Callable[..., Awaitable[dict]] = default_handler,

